src/CashuWallet.ts

```diff
@@ -327,15 +347,30 @@ class CashuWallet {
 			);
 			options?.proofsWeHave?.push(...keepProofsSelect);
 
-			const { keep, send } = await this.swap(amount, sendProofs, options);
-			const keepProofs = keepProofsSelect.concat(keep);
-			return { keep: keepProofs, send };
+			let { keep, send } = await this.swap(amount, sendProofs, options);
+			keep = keepProofsSelect.concat(keep);
+
+			// strip dleq if explicitly told so
```

Comment:
does `!options?.includeDleq` have two meanings in this case? Ignoring proofs without dleqs when `true` and stripping dleqs when `false` ? Maybe we should not overload this option to be more clear.

Reply:
Yea that's what I intended and what made most sense to me.

Reply:
Okay I see. I think most people will assume that not-passing an optional flag is equal to setting the boolean to false. Diverging from that needs to be documented very clearly I believe. However I would love to get some implementers thoughts on this. @callebtc @gudnuf could you leave your 2 sats on this?

Reply:
Could you maybe explain the usecase to me? WWhy would someone decide to strip DLEQs when sending to another wallet?